repos/system_upgrade/common/actors/trackcustommodifications/libraries/trackcustommodifications.py

```diff
@@ -0,0 +1,92 @@
+import ast
+import os
+
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.common.config.version import get_source_major_version, get_target_major_version
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import CustomModifications
+
+RPMS_TO_CHECK = ['leapp-upgrade-el{source}toel{target}']
+DIRS_TO_SCAN = ['/usr/share/leapp-repository']
+
+
+def _get_rpms_to_check():
+    return [rpm.format(source=get_source_major_version(), target=get_target_major_version()) for rpm in RPMS_TO_CHECK]
+
+
+def deduce_actor_name(a_file):
+    """A helper to map an actor/library to the actor name"""
+    if not os.path.exists(a_file) or not a_file.endswith('.py'):
+        return ''
+    data = None
+    with open(a_file) as f:
+        try:
+            data = ast.parse(f.read())
+        except TypeError:
+            api.current_logger().warning('An error occurred while parsing %s, can not deduce actor name', a_file)
+            return ''
+    # NOTE(ivasilev) Making proper syntax analysis is not the goal here, so let's get away with the bare minimum.
+    # An actor file will have an Actor ClassDef with a name attribute and at least one function defined
+    actor = next((obj for obj in data.body if isinstance(obj, ast.ClassDef) and obj.name and
+                  any(isinstance(o, ast.FunctionDef) for o in obj.body)), None)
```

Comment:
There are some false positives to this
```
/usr/share/leapp-repository/repositories/system_upgrade/common/actors/peseventsscanner/libraries/pes_event_parsing.py -> Package
/usr/share/leapp-repository/repositories/system_upgrade/common/actors/peseventsscanner/libraries/peseventsscanner_repomap.py -> RepoMapDataHandler
/usr/share/leapp-repository/repositories/system_upgrade/common/actors/repositoriesmapping/libraries/repositoriesmapping.py -> RepoMapData
/usr/share/leapp-repository/repositories/system_upgrade/common/actors/setuptargetrepos/libraries/setuptargetrepos_repomap.py -> RepoMapDataHandler
/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py -> _InputData
/usr/share/leapp-repository/repositories/system_upgrade/common/files/rhel_upgrade.py -> RhelUpgradeCommand
/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/dnfplugin.py -> _DnfPluginPathStr
/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/mounting.py -> MountError
/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/multipathutil.py -> LineData
/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/testutils.py -> produce_mocked
/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/config/version.py -> _SupportedVersionsDict
/usr/share/leapp-repository/repositories/system_upgrade/el8toel9/actors/nisscanner/libraries/nisscan.py -> NISScanLibrary
/usr/share/leapp-repository/repositories/system_upgrade/el8toel9/actors/sssdfacts/libraries/sssdfacts8to9.py -> SSSDFactsLibrary
```

However, an actor should have defined a ``produce`` function so adding a quick fix

```suggestion
                  any(isinstance(o, ast.FunctionDef) and o.name == 'produce' for o in obj.body)), None)
```

fixes most of them and then only remaining one is
```
/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/testutils.py -> CurrentActorMocked
```
which I guess it's a bit debatable whether it is a false positive or not.

Reply:
@dkubek Thanks a lot for catching this! Btw, what are your thoughts about tracking changes in actor libraries and tying them to the actor?

Reply:
I have fixed the parsing and added the majority of discovered false-positives as unit tests.